pkg/operator/starter.go

```diff
@@ -165,6 +168,27 @@ type controllerTemplateData struct {
 	CABundleConfigMap string
 }
 
+// withDeploymentStrategyHook sets the deployment.Spec.Strategy to Recreate on single-node deployments.
+// This is required to avoid deployment rollouts getting stuck.
+func withDeploymentStrategyHook(
+	configinformers configinformers.SharedInformerFactory,
+) csidrivercontrollerservicecontroller.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		infra, err := configinformers.Config().V1().Infrastructures().Lister().Get(infraConfigName)
+		if err != nil {
+			return err
+		}
+
+		// SingleReplicaTopologyMode is used in single-node deployments
+		if infra.Status.ControlPlaneTopology == v1.SingleReplicaTopologyMode {
+			deployment.Spec.Strategy = appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			}
+		}
+		return nil
+	}
+}
+
```

Comment:
AWS is the only operator that needs this for now, so I haven't moved this library-go yet.